os/binfmt/libelf/libelf_ctors.c

```diff
@@ -177,10 +177,9 @@ int elf_loadctors(FAR struct elf_loadinfo_s *loadinfo)
 
 #ifdef CONFIG_APP_BINARY_SEPARATION
 #ifdef CONFIG_DEBUG_MM_HEAPINFO
-			ARCH_GET_RET_ADDRESS
-			loadinfo->ctoralloc = (binfmt_ctor_t *)mm_malloc(loadinfo->uheap, ctorsize, retaddr);
+			loadinfo->ctoralloc = (binfmt_ctor_t *)malloc_user_at(loadinfo->uheap, ctorsize, retaddr);
 #else
-			loadinfo->ctoralloc = (binfmt_ctor_t *)mm_malloc(loadinfo->uheap, ctorsize);
+			loadinfo->ctoralloc = (binfmt_ctor_t *)malloc_user_at(loadinfo->uheap, ctorsize);
 #endif
```

Comment:
You should remove other lines to leave only below
```
loadinfo->ctoralloc = (binfmt_ctor_t *)malloc_user_at(loadinfo->uheap, ctorsize);
```

Reply:
Modified. Thanks.